docs/types/literal-suffixes.rst

```diff
@@ -0,0 +1,243 @@
+.. index:: ! literal suffix
+.. _literal_suffixes:
+
+Literal Suffixes
+================
+
+While Solidity :ref:`provides implicit conversions<types-conversion-literals>` from its literals to selected types,
+there are types for which no built-in conversions are available (most notably the
+:ref:`user-defined value types<user-defined-value-types>`).
+
+To fill this gap, it is possible to define custom conversions in the form of *literal suffixes*.
+
+.. code-block:: solidity
+    :force:
+
+    // SPDX-License-Identifier: GPL-3.0
+    pragma solidity ^0.8.20;
+
+    type Coin is uint;
+    using {add as +, eq as ==} for Coin global;
+
+    function add(Coin a, Coin b) pure returns (Coin) {
+        return Coin.wrap(Coin.unwrap(a) + Coin.unwrap(b));
+    }
+
+    function eq(Coin a, Coin b) pure returns (bool) {
+        return Coin.unwrap(a) == Coin.unwrap(b);
+    }
+
+    function c(uint mantissa, uint exponent) pure suffix returns (Coin) {
+        return Coin.wrap(mantissa * 10**6 / 10**exponent);
+    }
+
+    function uc(uint microValue) pure suffix returns (Coin) {
+        return Coin.wrap(microValue);
+    }
+
+    contract CoinSeller {
+        mapping(address => Coin) balances;
+
+        function buy() public payable {
+            require(msg.value == 1 ether);
+            assert(1.5 c == 1_500_000 uc);
+            balances[msg.sender] = balances[msg.sender] + 1.5 c;
+        }
+    }
+
+.. index:: ! literal suffix;definition, function;free
+
+Defining Suffix Functions
+-------------------------
+
+Literal suffixes can be defined by applying the built-in ``suffix`` modifier to a :ref:`free function<functions>`.
+
+Only pure functions can be used as suffixes.
+This means that suffixes cannot read or modify contract state.
+As all pure functions, however, they can perform pure external calls.
```

Comment:
```suggestion
Literal suffixes can be defined by applying the built-in ``suffix`` modifier to a :ref:`free function<functions>`.

Only pure functions can be used as suffixes.
This means that suffixes cannot read or modify contract states.
As with all pure functions, however, they can perform pure external calls.
```

Reply:
Contract state would be preferred (for me at least), as all of the contract's storage variables make up its state, rather than referring to each individual storage variable as a state.

Reply:
`contract states` sounds weird to me. I usually see it referred to "contract sate".

I just checked the docs and looks like we never really say specifically "contract state" there. Only "blockchain state". Changed to that.